org.osgi.service.feature/src/org/osgi/service/feature/FeatureBuilder.java

```diff
@@ -113,20 +113,27 @@
     FeatureBuilder addExtensions(FeatureExtension ... extensions);
 
     /**
-     * Add a variable to the Feature. If a variable with the specified key
-     * already exists it is replaced with this one.
-     * @param key The key.
-     * @param defaultValue The default value.
-     * @return This builder.
-     */
-    FeatureBuilder addVariable(String key, String defaultValue);
+	 * Add a variable to the Feature. If a variable with the specified key
+	 * already exists it is replaced with this one. Variable values are of type:
+	 * String, Boolean or BigDecimal for numbers.
+	 * 
+	 * @param key The key.
+	 * @param defaultValue The default value.
+	 * @return This builder.
+	 * @throws IllegalArgumentException if the value is of an invalid type.
+	 */
+	FeatureBuilder addVariable(String key, Object defaultValue);
```

Comment:
maybe just `value` not `defaultValue`

Reply:
ignore that

Reply:
Yes, variables have a 'default' value described in the feature. This value is used by a launcher if no value is provided externally.